xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +196,215 @@ private static ManagedChannel initLbChannel(
+    }
+
+    final class FixedXdsConfigBalancer extends LoadBalancer {
+      final Helper helper;
+
+      // All fields become non-null once handleResolvedAddresses() successfully.
+      // All fields are assigned at most once.
+      @Nullable
+      ObjectPool<XdsClient> xdsClientRef;
+      @Nullable
+      XdsClient xdsClient;
+      @Nullable
+      LocalityStore localityStore;
+      @Nullable
+      LoadReportClient lrsClient;
+
+      FixedXdsConfigBalancer(Helper helper) {
+        this.helper = helper;
+      }
+
+      @Override
+      public void handleNameResolutionError(Status error) {}
+
+      @Override
+      public boolean canHandleEmptyAddressListFromNameResolution() {
+        return true;
+      }
+
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        if (xdsClientRef != null) {
+          // handleResolvedAddresses() already called, so no-op here, because the config is fixed in
+          // this balancer.
+          return;
+        }
+
+        // TODO(zdapeng): First try to get loadStatsStore from XdsAttributes.LOAD_STATS_STORE_REF.
+        LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+        localityStore = localityStoreFactory.newLocalityStore(
+            helper, lbRegistry, loadStatsStore);
+        // TODO(zdapeng): Use XdsClient to do Lrs directly.
+        final LoadReportCallback lrsCallback =
+            new LoadReportCallback() {
+              @Override
+              public void onReportResponse(long reportIntervalNano) {
+                localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+              }
+            };
+
+        // There are three usecases:
+        // 1. The EDS-only legacy usecase that requires balancerName from xds config.
+        //    TODO(zdapeng): Remove the legacy case.
+        // 2. The EDS-only with bootstrap usecase:
+        //    The name resolver resolves a ResolvedAddresses with an XdsConfig without balancerName
+        //    field. Use the bootstrap information to create a channel.
+        // 3. Non EDS-only usecase:
+        //    XDS_CLIENT_REF attribute is available from ResolvedAddresses either from
+        //    XdsNameResolver or CDS policy.
+        // We assume XdsConfig switching happens only within one usecase, and there is no switching
+        // between different usecases.
+        if (xdsConfig.balancerName != null) {
+          // This is the EDS-only legacy usecase that requires balancerName from xds config.
+
+          if (oldXdsConfig == null || !xdsConfig.balancerName.equals(oldXdsConfig.balancerName)) {
+            final ManagedChannel channel = initLbChannel(
+                helper, xdsConfig.balancerName, Collections.<ChannelCreds>emptyList());
+            final Node node = Node.newBuilder()
+                .setMetadata(Struct.newBuilder()
+                    .putFields(
+                        "endpoints_required",
```

Comment:
Do we still need this? If not, just eliminate it.

Reply:
Done.